### PR TITLE
padding + margin added to layout components

### PR DESCRIPTION
**Pull request by michelengelen**
#### Summary
introduces new globally shared props types `PGlobalsMisc`, `PGlobalsLayout` and (the previous two combined) `PGlobals` for better/easier props definition with shared props like `className`, `children`, `padding` and `margin`

### CODE REVIEW

File: src/shared/props.ts [hunk 1]
```diff
@@ -243,7 +247,46 @@ type PAriaAllAttributes = PAriaWidgetAttributes &
...
+ */
+type PGlobalsMisc = {
+    /**
+     * custom className
+     */
+    className?: string;
+    /**
+     * component children
+     */
+    children?: ReactNode | ReactNodeArray;
```
Comment (nevyangelova) [inline src/shared/props.ts:281]: what about adding event handles and aria props here as well?

**Review by nevyangelova — COMMENTED** [review]
LGTM

File: src/shared/props.ts (hunk 1 above)
Comment (michelengelen) [reply src/shared/props.ts:281]: Event handlers could be an option, but depending on the situation a handling function can have a totally different structure

**Comment by michelengelen** [comment]
@nevyangelova kind reminder to re-review

**Review by nevyangelova — APPROVED** [review]

**Pull request merged**